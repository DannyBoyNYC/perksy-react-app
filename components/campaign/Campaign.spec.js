const React = require('react');
const { mount, shallow } = require('enzyme');

const Campaign = require('./index').default;

describe('<Campaign.* />', () => {
  let component;
  const requiredProps = {
    name: 'Campaign Name',
    respondents: {
      current: 898,
      maximum: 1112
    }
  };

  describe('<Campaign.Card />', () => {
    beforeEach(() => {
      component = shallow(<Campaign.Card {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('campaign campaign--card');
    });
  });

  describe('<Campaign.Progress />', () => {
    beforeEach(() => {
      component = shallow(<Campaign.Progress {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('campaign campaign--progress');
    });

    it('should render a ProgressBar', () => {
      component = mount(<Campaign.Progress {...requiredProps} />);
      expect(component.find('ProgressBar')).toExist();
    });
  });
});
