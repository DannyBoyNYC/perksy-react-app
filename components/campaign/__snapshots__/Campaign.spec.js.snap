// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Campaign.* /> <Campaign.Card /> should render correctly with all required props 1`] = `
<div
  className="campaign campaign--card"
>
  <div
    className="campaign__image"
  />
  <div
    className="campaign__name"
  >
    Campaign Name
  </div>
  <div
    className="campaign__additional-components"
  />
</div>
`;

exports[`<Campaign.* /> <Campaign.Progress /> should render correctly with all required props 1`] = `
<div
  className="campaign campaign--progress"
>
  <div
    className="campaign__image"
  />
  <div
    className="campaign__name"
  >
    Campaign Name
  </div>
  <div
    className="campaign__additional-components"
  />
  <div
    className="campaign__progress"
  >
    <Basic
      current={898}
      maximum={1112}
    />
  </div>
</div>
`;
